build script for leveldbjni

asellappen opened the pull request.

asellappen commented:
[INFO] --- maven-clean-plugin:2.3:clean (default-clean) @ leveldbjni-all ---
[INFO]
[INFO] --- maven-resources-plugin:2.5:resources (default-resources) @ leveldbjni-all ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /leveldbjni/leveldbjni-all/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:2.3.2:compile (default-compile) @ leveldbjni-all ---
[INFO] Compiling 1 source file to /leveldbjni/leveldbjni-all/target/classes
[INFO]
[INFO] --- maven-resources-plugin:2.5:testResources (default-testResources) @ leveldbjni-all ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /leveldbjni/leveldbjni-all/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:2.3.2:testCompile (default-testCompile) @ leveldbjni-all ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:2.4.3:test (default-test) @ leveldbjni-all ---
[INFO] No tests to run.
[INFO]
[INFO] --- maven-bundle-plugin:2.3.7:bundle (default-bundle) @ leveldbjni-all ---
[WARNING] Bundle org.fusesource.leveldbjni:leveldbjni-all:bundle:1.8 : Split package org/fusesource/leveldbjni
Use directive -split-package:=(merge-first|merge-last|error|first) on Export/Private Package instruction to get rid of this warning
Package found in   [Jar:., Jar:leveldbjni]
Reference from     /leveldbjni/leveldbjni/target/leveldbjni-1.8.jar
Classpath          [Jar:., Jar:leveldbjni, Jar:hawtjni-runtime, Jar:leveldb-api, Jar:leveldbjni-osx, Jar:leveldbjni-linux32, Jar:leveldbjni-linux64, Jar:leveldbjni-win32, Jar:leveldbjni-win64]
[INFO]
[INFO] --- maven-install-plugin:2.3.1:install (default-install) @ leveldbjni-all ---
[INFO] Installing /leveldbjni/leveldbjni-all/target/leveldbjni-all-1.8.jar to /root/.m2/repository/org/fusesource/leveldbjni/leveldbjni-all/1.8/leveldbjni-all-1.8.jar
[INFO] Installing /leveldbjni/leveldbjni-all/pom.xml to /root/.m2/repository/org/fusesource/leveldbjni/leveldbjni-all/1.8/leveldbjni-all-1.8.pom
[INFO]
[INFO] --- maven-bundle-plugin:2.3.7:install (default-install) @ leveldbjni-all ---
[INFO] Writing OBR metadata
[INFO] Installing org/fusesource/leveldbjni/leveldbjni-all/1.8/leveldbjni-all-1.8.jar
[INFO] Writing OBR metadata
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for leveldbjni-project 1.8:
[INFO]
[INFO] leveldbjni-project ................................. SUCCESS [ 24.384 s]
[INFO] leveldbjni ......................................... SUCCESS [01:38 min]
[INFO] leveldbjni-all ..................................... SUCCESS [  5.206 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:19 min
[INFO] Finished at: 2021-04-19T16:44:28Z
[INFO] ------------------------------------------------------------------------
root@07e81a131b7c:/#

asellappen commented:
The following jar files are created as expected (leveldbjni-all-1.8.jar)

root@07e81a131b7c:/leveldbjni/leveldbjni-all/target# ls -ltr
total 1032
drwxr-xr-x 3 root root    4096 Apr 19 16:44 generated-sources
-rw-r--r-- 1 root root 1045791 Apr 19 16:44 leveldbjni-all-1.8.jar
drwxr-xr-x 4 root root    4096 Apr 19 16:44 classes
root@07e81a131b7c:/leveldbjni/leveldbjni-all/target#

root@07e81a131b7c:/leveldbjni/leveldbjni/target# ls -ltr
total 472
drwxr-xr-x 3 root root   4096 Apr 19 16:43 generated-sources
drwxr-xr-x 4 root root   4096 Apr 19 16:43 classes
drwxr-xr-x 3 root root   4096 Apr 19 16:43 test-classes
drwxr-xr-x 2 root root   4096 Apr 19 16:44 maven-archiver
-rw-r--r-- 1 root root  56412 Apr 19 16:44 leveldbjni-1.8.jar
-rw-r--r-- 1 root root  11743 Apr 19 16:44 leveldbjni-1.8-tests.jar
-rw-r--r-- 1 root root 349533 Apr 19 16:44 leveldbjni-1.8-native-src.zip
-rw-r--r-- 1 root root  43126 Apr 19 16:44 leveldbjni-1.8-sources.jar
root@07e81a131b7c:/leveldbjni/leveldbjni/target#

asellappen commented:
@seth-priya / @gururajrkatti , Can you review this PR ?

gururajrkatti commented on l/leveldbjni/leveldbjni_ubuntu_18.04.sh:
```diff
@@ -0,0 +1,92 @@
+wget http://repository.timesys.com/buildsources/s/snappy/snappy-1.0.5/snappy-1.0.5.tar.gz
+tar -zxvf snappy-1.0.5.tar.gz
+git clone git://github.com/chirino/leveldb.git
+git clone git://github.com/fusesource/leveldbjni.git
```
Is it not already cloned using "git clone ${REPO}" ?

asellappen replied:
thanks for pointing this, wrongly done again 
.removed in the latest script and tested .

gururajrkatti commented on l/leveldbjni/leveldbjni_ubuntu_18.04.sh:
```diff
@@ -0,0 +1,92 @@
+
+
+cd ${LEVELDBJNI_HOME}
+cd ..
```
What is the reason for changing to LEVELDBJNI_HOME and then to its parent directory ?

asellappen replied:
pom.xml for creating leveldbjni-all-1.8.jar is in the parent directory ,hence moving to there .

During the  below steps, The execution  pointer is moving to different paths based on the script written by community ,

cd ${SNAPPY_HOME}
./configure --build ppc64le --disable-shared --with-pic
make
cd ${LEVELDB_HOME}
export LIBRARY_PATH=${SNAPPY_HOME}
export C_INCLUDE_PATH=${LIBRARY_PATH}
export CPLUS_INCLUDE_PATH=${LIBRARY_PATH}
git apply ../leveldb.patch
make libleveldb.a

To process the pom.xml required to generate leveldbjni-all is at this path hence moving to parent dir as said in the script.

gururajrkatti commented on l/leveldbjni/leveldbjni_ubuntu_18.04.sh:
```diff
@@ -0,0 +1,92 @@
+# ----------------------------------------------------------------------------
+#
+# Package          : leveldbjni
+# Version          : 0.1.8
```
Is it "1.8" or "0.1.8" ?

asellappen replied:
changed to 1.8

asellappen commented:
changed the version to 1.8 and removed git clone git://github.com/fusesource/leveldbjni.git( extra  clone wrongly done)

 **sh leveldbjni_ubuntu_18.04.sh**

[INFO] Building jar: /leveldbjni/leveldbjni/target/leveldbjni-1.8-sources.jar
[INFO]
[INFO] --- maven-install-plugin:2.3.1:install (default-install) @ leveldbjni ---
[INFO] Installing /leveldbjni/leveldbjni/target/leveldbjni-1.8.jar to /root/.m2/repository/org/fusesource/leveldbjni/leveldbjni/1.8/leveldbjni-1.8.jar
[INFO] Installing /leveldbjni/leveldbjni/pom.xml to /root/.m2/repository/org/fusesource/leveldbjni/leveldbjni/1.8/leveldbjni-1.8.pom
[INFO] Installing /leveldbjni/leveldbjni/target/leveldbjni-1.8-tests.jar to /root/.m2/repository/org/fusesource/leveldbjni/leveldbjni/1.8/leveldbjni-1.8-tests.jar
[INFO] Installing /leveldbjni/leveldbjni/target/leveldbjni-1.8-native-src.zip to /root/.m2/repository/org/fusesource/leveldbjni/leveldbjni/1.8/leveldbjni-1.8-native-src.zip
[INFO] Installing /leveldbjni/leveldbjni/target/leveldbjni-1.8-sources.jar to /root/.m2/repository/org/fusesource/leveldbjni/leveldbjni/1.8/leveldbjni-1.8-sources.jar
[INFO]
[INFO] --------------< org.fusesource.leveldbjni:leveldbjni-all >--------------
[INFO] Building leveldbjni-all 1.8                                        [3/3]
[INFO] -------------------------------[ bundle ]-------------------------------
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-osx/1.8/leveldbjni-osx-1.8.pom
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-osx/1.8/leveldbjni-osx-1.8.pom
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-osx/1.8/leveldbjni-osx-1.8.pom (4.1 kB at 14 kB/s)
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-linux32/1.8/leveldbjni-linux32-1.8.pom
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-linux32/1.8/leveldbjni-linux32-1.8.pom
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-linux32/1.8/leveldbjni-linux32-1.8.pom (3.9 kB at 23 kB/s)
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-linux64/1.8/leveldbjni-linux64-1.8.pom
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-linux64/1.8/leveldbjni-linux64-1.8.pom
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-linux64/1.8/leveldbjni-linux64-1.8.pom (3.8 kB at 12 kB/s)
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-win32/1.8/leveldbjni-win32-1.8.pom
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-win32/1.8/leveldbjni-win32-1.8.pom
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-win32/1.8/leveldbjni-win32-1.8.pom (3.8 kB at 12 kB/s)
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-win64/1.8/leveldbjni-win64-1.8.pom
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-win64/1.8/leveldbjni-win64-1.8.pom
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-win64/1.8/leveldbjni-win64-1.8.pom (3.8 kB at 12 kB/s)
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-osx/1.8/leveldbjni-osx-1.8.jar
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-linux32/1.8/leveldbjni-linux32-1.8.jar
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-linux64/1.8/leveldbjni-linux64-1.8.jar
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-win32/1.8/leveldbjni-win32-1.8.jar
Downloading from fusesource.nexus.snapshot: http://repo.fusesource.com/nexus/content/groups/public-snapshots/org/fusesource/leveldbjni/leveldbjni-win64/1.8/leveldbjni-win64-1.8.jar
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-osx/1.8/leveldbjni-osx-1.8.jar
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-linux32/1.8/leveldbjni-linux32-1.8.jar
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-win64/1.8/leveldbjni-win64-1.8.jar
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-win32/1.8/leveldbjni-win32-1.8.jar
Downloading from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-linux64/1.8/leveldbjni-linux64-1.8.jar
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-linux32/1.8/leveldbjni-linux32-1.8.jar (284 kB at 1.3 MB/s)
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-osx/1.8/leveldbjni-osx-1.8.jar (170 kB at 495 kB/s)
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-win64/1.8/leveldbjni-win64-1.8.jar (128 kB at 372 kB/s)
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-win32/1.8/leveldbjni-win32-1.8.jar (113 kB at 174 kB/s)
Downloaded from sonatype-nexus: https://oss.sonatype.org/content/repositories/public/org/fusesource/leveldbjni/leveldbjni-linux64/1.8/leveldbjni-linux64-1.8.jar (282 kB at 416 kB/s)
[INFO]
[INFO] --- maven-clean-plugin:2.3:clean (default-clean) @ leveldbjni-all ---
[INFO]
[INFO] --- maven-resources-plugin:2.5:resources (default-resources) @ leveldbjni-all ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /leveldbjni/leveldbjni-all/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:2.3.2:compile (default-compile) @ leveldbjni-all ---
[INFO] Compiling 1 source file to /leveldbjni/leveldbjni-all/target/classes
[INFO]
[INFO] --- maven-resources-plugin:2.5:testResources (default-testResources) @ leveldbjni-all ---
[debug] execute contextualize
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /leveldbjni/leveldbjni-all/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:2.3.2:testCompile (default-testCompile) @ leveldbjni-all ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:2.4.3:test (default-test) @ leveldbjni-all ---
[INFO] No tests to run.
[INFO]
[INFO] --- maven-bundle-plugin:2.3.7:bundle (default-bundle) @ leveldbjni-all ---
[WARNING] Bundle org.fusesource.leveldbjni:leveldbjni-all:bundle:1.8 : Split package org/fusesource/leveldbjni
Use directive -split-package:=(merge-first|merge-last|error|first) on Export/Private Package instruction to get rid of this warning
Package found in   [Jar:., Jar:leveldbjni]
Reference from     /leveldbjni/leveldbjni/target/leveldbjni-1.8.jar
Classpath          [Jar:., Jar:leveldbjni, Jar:hawtjni-runtime, Jar:leveldb-api, Jar:leveldbjni-osx, Jar:leveldbjni-linux32, Jar:leveldbjni-linux64, Jar:leveldbjni-win32, Jar:leveldbjni-win64]
[INFO]
[INFO] --- maven-install-plugin:2.3.1:install (default-install) @ leveldbjni-all ---
[INFO] Installing /leveldbjni/leveldbjni-all/target/leveldbjni-all-1.8.jar to /root/.m2/repository/org/fusesource/leveldbjni/leveldbjni-all/1.8/leveldbjni-all-1.8.jar
[INFO] Installing /leveldbjni/leveldbjni-all/pom.xml to /root/.m2/repository/org/fusesource/leveldbjni/leveldbjni-all/1.8/leveldbjni-all-1.8.pom
[INFO]
[INFO] --- maven-bundle-plugin:2.3.7:install (default-install) @ leveldbjni-all ---
[INFO] Writing OBR metadata
[INFO] Installing org/fusesource/leveldbjni/leveldbjni-all/1.8/leveldbjni-all-1.8.jar
[INFO] Writing OBR metadata
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for leveldbjni-project 1.8:
[INFO]
[INFO] leveldbjni-project ................................. SUCCESS [ 21.720 s]
[INFO] leveldbjni ......................................... SUCCESS [01:30 min]
[INFO] leveldbjni-all ..................................... SUCCESS [  5.211 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:59 min
[INFO] Finished at: 2021-05-07T10:01:25Z
[INFO] ------------------------------------------------------------------------
root@136e57d227ea:/# cd /leveldbjni/leveldbjni-all/target
root@136e57d227ea:/leveldbjni/leveldbjni-all/target# ls -ltr
total 1032
drwxr-xr-x 3 root root    4096 May  7 10:01 generated-sources
**-rw-r--r-- 1 root root 1045790 May  7 10:01 leveldbjni-all-1.8.jar**
drwxr-xr-x 4 root root    4096 May  7 10:01 classes
root@136e57d227ea:/leveldbjni/leveldbjni-all/target#



 **sh leveldbjni_ubuntu_18.04.sh 'leveldbjni-1.7'**
 
 
 NFO] --- maven-bundle-plugin:2.3.7:install (default-install) @ leveldbjni-all ---
[INFO] Installing org/fusesource/leveldbjni/leveldbjni-all/1.7/leveldbjni-all-1.7.jar
[INFO] Writing OBR metadata
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for leveldbjni-project 1.7:
[INFO]
[INFO] leveldbjni-project ................................. SUCCESS [  0.272 s]
[INFO] leveldbjni ......................................... SUCCESS [ 12.450 s]
[INFO] leveldbjni-all ..................................... SUCCESS [  8.219 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  22.036 s
[INFO] Finished at: 2021-05-07T10:09:07Z
[INFO] ------------------------------------------------------------------------

root@136e57d227ea:/leveldbjni/leveldbjni-all/target# ls -ltr
total 1020
drwxr-xr-x 3 root root    4096 May  7 10:09 generated-sources
**-rw-r--r-- 1 root root 1035514 May  7 10:09 leveldbjni-all-1.7.jar**
drwxr-xr-x 4 root root    4096 May  7 10:09 classes
root@136e57d227ea:/leveldbjni/leveldbjni-all/target#

gururajrkatti reviewed (approved):
ok
